tests/unit/geo.tcl

```
@@ -166,6 +166,66 @@ start_server {tags {"geo"}} {
         r georadius nyc -73.9798091 40.7598464 3 km asc
     } {{central park n/q/r} 4545 {union square}}
 
+    test {GEOSEARCH with wrong type src key should throw error} {
+        r set src{t} wrong_type
+
+        assert_error "WRONGTYPE*" {r geosearch src{t} fromlonlat 0 0 byradius 1 km asc}
+        assert_error "WRONGTYPE*" {r geosearch src{t} fromlonlat 0 0 byradius 1 km withdist desc count 1}
+        assert_error "WRONGTYPE*" {r geosearch src{t} fromlonlat 0 0 byradius 1 km withcoord asc count 1 any}
+
+        assert_error "WRONGTYPE*" {r geosearch src{t} frommember member bybox 1 1 km asc}
+        assert_error "WRONGTYPE*" {r geosearch src{t} frommember member bybox 1 1 km asc count 1}
+        assert_error "WRONGTYPE*" {r geosearch src{t} frommember member bybox 1 1 km withhash asc count 1 any}
+    }
+
+    test {GEOSEARCH with non existing src key should return emptyarray} {
+        r del src{t}
+
+        # Make sure we can distinguish between an empty array and a null response
+        r readraw 1
```

i don't think we wanna mess with raw protocol in any of these tests.
the bug was returning an integer rather than an array, no need for low level protocol to make that distinction (or detect any similar future breakage).

the reason i asked for so many tests was because the code is a bit convoluted, there are several different commands routed into the same function and several different ways it can conclude it's a STORE variant.
and sometimes we only realize we're a STORE variant after already running into some error (but still we need to return the proper error the client expects)...

but none of that requires low level protocol checks IMHO